inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -827,8 +911,35 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
     auto expected_tensor = current_layout.size;
     auto expected_data_type = current_layout.data_type;
     auto expected_format = current_layout.format;
-
-    if (_optimization_attributes.b_fs_zyx_fsv16_network &&
+    auto input_layout = node.get_dependency(0).get_output_layout();
+    auto is_2d = input_layout.format.spatial_num() == 2;
+    bool use_onednn_impls = _optimization_attributes.use_onednn_impls;
+
+    bool onednn_valid_dt = input_layout.data_type == data_types::i8 ||
+                           input_layout.data_type == data_types::u8 ||
+                           input_layout.data_type == data_types::f16;
+
+    bool onednn_valid_params = onednn_valid_dt &&
+                               input_layout.size.feature[0] >= 16 &&
+                               prim->groups == 1 &&
+                               /* get_post_ops_count(node) <= 10 */ false &&
+                               input_layout.size.batch[0] < 16;  // oneDNNs optimized kernel doesn't support big batches yet
+
+    if (use_onednn_impls && onednn_valid_params) {
+        if (input_layout.data_type == data_types::f16) {
+            if (input_layout.size.batch[0] < 16) {
+                expected_format = is_2d ? cldnn::format::b_fs_yx_fsv16 : cldnn::format::b_fs_zyx_fsv16;
```

Comment:
What about `wzyx`? I think we need to add some assertion code to confirm that it is either yx or zyx, not other.